File: src/main/java/com/github/unldenis/hologram/line/animated/IAnimated.java
package com.github.unldenis.hologram.line.animated;

import com.github.unldenis.hologram.Hologram;
import com.github.unldenis.hologram.animation.Animation;
import java.util.Collection;
import java.util.Optional;
import org.bukkit.entity.Player;

public interface IAnimated {
   void setAnimation(Animation animation, Collection<Player> seeingPlayers);

   default void setAnimation(Animation animation, Hologram hologram) {
      setAnimation(animation, hologram.getSeeingPlayers());
   }

   default void setAnimation(Animation.AnimationType animationType, Collection<Player> seeingPlayers) {
      setAnimation(animationType.type.get(), seeingPlayers);
   }

   default void setAnimation(Animation.AnimationType animationType, Hologram hologram) {
      setAnimation(animationType, hologram.getSeeingPlayers());
   }

   void removeAnimation();

   Optional<Animation> getAnimation();
}

File: src/main/java/com/github/unldenis/hologram/line/animated/IAnimated.kt
package com.github.unldenis.hologram.line.animated

import com.github.unldenis.hologram.animation.Animation
import com.github.unldenis.hologram.animation.AnimationType
import com.github.unldenis.hologram.hologram.Hologram
import org.bukkit.entity.Player

interface IAnimated {
    fun setAnimation(animation: Animation, seeingPlayers: Collection<Player>)

    fun setAnimation(animation: Animation, hologram: Hologram) {
        setAnimation(animation, hologram.seeingPlayers)
    }

    fun setAnimation(animationType: AnimationType, seeingPlayers: Collection<Player>) {
        setAnimation(animationType.type.get(), seeingPlayers)
    }

    fun setAnimation(animationType: AnimationType, hologram: Hologram) {
        setAnimation(animationType, hologram.seeingPlayers)
    }

    fun removeAnimation()

    var animation: Animation
}
